src/compiler/emitter.ts

```diff
@@ -4530,7 +4530,7 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                 write("class");
 
                 // check if this is an "export default class" as it may not have a name. Do not emit the name if the class is decorated.
-                if ((node.name || !(node.flags & NodeFlags.Default)) && !thisNodeIsDecorated) {
+                if ((node.name || (node.flags & NodeFlags.Default)) && !thisNodeIsDecorated) {

```

Comment:
Yes, currently it should and personally I don't like it - having multiple ways to check target language version  is confusing especially because correct one is non-obvious. I would honestly prefer that `createProgram` would clone compilerOptions argument and then augment it with default values so we can just check `CompilerOptions.target` everywhere